scylla-cdc-replicator/src/test/java/com/scylladb/cdc/replicator/ReplicatorCorrectnessTestIT.java

```diff
@@ -0,0 +1,616 @@
+            with().pollInterval(200, TimeUnit.MILLISECONDS).await()
+                    .atMost(DEFAULT_AWAIT_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+      private Session srcSession;
+      private Session dstSession;
+
+
+      @BeforeAll
+      public void prepare() {
+            srcSession = buildLibrarySessionSrc();
+            dstSession = buildLibrarySessionDst();
+
+            for (String query : UDTsCreate) {
+                  srcSession.execute(query);
+                  dstSession.execute(query);
+            }
+      }
+
+      static class Schema {
+            public String keyspace;
+            public String tableName;
+            public String createTable;
+
+            public Schema(String keyspace, String tableName, String createTable) {
+                  this.keyspace = keyspace;
+                  this.tableName = tableName;
+                  this.createTable = createTable;
+            }
+      }
+
+      static class TestData {
+            public String testName;
+            public Schema schema;
+            public String[] queries;
+
+            public TestData(String testName, Schema schema, String[] queries) {
+                  this.testName = testName;
+                  this.schema = schema;
+                  this.queries = queries;
+            }
+      }
+
+      private void prepareTables(TestData data) {
+            srcSession.execute(SchemaBuilder.dropTable(data.schema.keyspace, data.schema.tableName).ifExists());
+            dstSession.execute(SchemaBuilder.dropTable(data.schema.keyspace, data.schema.tableName).ifExists());
+
+            srcSession.execute(data.schema.createTable + " WITH cdc = {'enabled': true, 'preimage': true, 'postimage': true}");
+            dstSession.execute(data.schema.createTable);
+      }
+
+      private Optional<String> verifyResult(Select select_all) {
+            ResultSet srcRes = srcSession.execute(select_all);
+            ResultSet dstRes = dstSession.execute(select_all);
+            List<Row> srcRows = srcRes.all();
+            List<Row> dstRows = dstRes.all();
+
+            if (srcRows.size() != dstRows.size()) {
+                  return Optional.of(String.format("Row amount not equal. Source: %d, destination: %d", srcRows.size(), dstRows.size()));
+            }
+
+            Iterator<Row> srcIt = srcRows.iterator();
+            Iterator<Row> dstIt = dstRows.iterator();
+
+            while(srcIt.hasNext() && dstIt.hasNext()) {
+                  Row srcRow = srcIt.next();
+                  Row dstRow = dstIt.next();
+                  Optional<Definition> wrongColumnOptional = srcRow.getColumnDefinitions().asList().stream().filter(def -> {
+                        ByteBuffer srcBytes = srcRow.getBytesUnsafe(def.getName());
+                        ByteBuffer dstBytes = dstRow.getBytesUnsafe(def.getName());
+                        if (srcBytes == null) {
+                              return srcBytes != null;
```

Comment:
Should be: `dstBytes != null`. Even better to use: `Objects.equals(srcBytes, dstBytes)` which is `null`-safe.